src/reify.cpp

```diff
@@ -0,0 +1,333 @@
+//
+// Copyright (c) 2017 - 2025, Roland Kaminski
+// Copyright (c) 2025 - present, Francois Laferriere
+//
+// This file is part of Potassco.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+#include <potassco/enum.h>
+#include <potassco/error.h>
+#include <potassco/graph.h>
+#include <potassco/reify.h>
+
+#include <algorithm>
+#include <ostream>
+#include <unordered_map>
+
+namespace Potassco {
+namespace {
+/////////////////////////////////////////////////////////////////////////////////////////
+// Helpers
+/////////////////////////////////////////////////////////////////////////////////////////
+struct Head {
+    HeadType type;
+    size_t   id;
+};
+
+struct Normal {
+    size_t id;
+};
+
+struct Sum {
+    size_t   id;
+    Weight_t bound;
+};
+
+struct Quoted {
+    std::string_view str;
+};
+
+template <typename T>
+void printValue(std::ostream& out, const T& value) {
+    out << value;
+}
+
+inline void printValue(std::ostream& out, const WeightLit& value) {
```

Comment:
Nitpick style: Please consider either dropping the `inline` or replace it with `POTASSCO_FORCE_INLINE`.
For ODR/visibility-reasons, the `inline` is **not** needed here, while all major compilers ignore the "performance-hint" character of plain `inline`.